File: python/7kyu/Complementary_DNA.py
def DNA_strand(dna):
    dna_list = list(dna)
    for i in range(len(dna_list)):
        if dna_list[i] == "A":
            dna_list[i] = "T"
        elif dna_list[i] == "T":
            dna_list[i] = "A"
        elif dna_list[i] == "C":
            dna_list[i] = "G"
        elif dna_list[i] == "G":
            dna_list[i] = "C"
    return "".join(dna_list)

File: typescript/7kyu/Complementary_DNA.ts
export class Kata {
  static dnaStrand(dna: string) {
    const dnaArr = dna.split("");
    dnaArr.forEach((el, idx) => {
      if (el === "A") dnaArr.splice(idx, 1, "T");
      if (el === "T") dnaArr.splice(idx, 1, "A");
      if (el === "C") dnaArr.splice(idx, 1, "G");
      if (el === "G") dnaArr.splice(idx, 1, "C");
    });
    return dnaArr.join("");
  }
}
